extensions-core/azure-extensions/src/main/java/org/apache/druid/storage/azure/AzureStorage.java

```diff
@@ -116,7 +116,7 @@ public void uploadBlockBlob(final File file, final String containerName, final S
     try (FileInputStream stream = new FileInputStream(file)) {
       // By default this creates a Block blob, no need to use a specific Block blob client.
       // We also need to urlEncode the path to handle special characters.
-      blobContainerClient.getBlobClient(Utility.urlEncode(blobPath)).upload(stream, file.length());
+      blobContainerClient.getBlobClient(Utility.urlEncode(blobPath)).upload(stream, file.length(), true);
```

Comment:
`overwrite=true` is now not required since we would not be. pushing files multiple times now ?

Reply:
i wanted to bring the azure client in line with how other deep storage systems handle this like s3/google cloud

Reply:
May be a comment would be good then.

Reply:
yeah makes sense, will add one